pkg/accumulator/accumulator.go

```diff
@@ -0,0 +1,256 @@
+package accumulator
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"sniffer/pkg/config"
+	"sniffer/pkg/ebpfeng"
+	evData "sniffer/pkg/ebpfev/v1"
+
+	logger "github.com/kubescape/go-logger"
+	"github.com/kubescape/go-logger/helpers"
+)
+
+const (
+	ACCUMULATOR_SIZE = 10
+)
+
+const (
+	DROP_EVENT_OCCURRED = "drop event occurred\n"
+)
+
+type containersAccumulator struct {
+	accumulatorDataPerContainer map[string]chan evData.EventData
+	registerMutex               sync.Mutex
+}
+
+type Accumulator struct {
+	accumulatorData                 []map[string][]evData.EventData
+	syncReaderWriterAccumulatorData sync.Mutex
+	firstMapKeysOfAccumulatorData   []string
+	cacheAccumulatorSize            int
+	mainDataChannel                 chan *evData.EventData
+	containersData                  containersAccumulator
+	ebpfEngine                      ebpfeng.EbpfEngineClient
+}
+
+type ContainerAccumulator struct {
+	dataChannel chan evData.EventData
+	containerID string
+}
+
+var myContainerID string
+var accumulatorSingelton *Accumulator
+var lock = &sync.Mutex{}
+
+func createAccumulator() *Accumulator {
+	myContainerID = "111111"
+	accumulatorSingelton = &Accumulator{
+		cacheAccumulatorSize:          ACCUMULATOR_SIZE,
+		accumulatorData:               make([]map[string][]evData.EventData, ACCUMULATOR_SIZE),
+		firstMapKeysOfAccumulatorData: make([]string, ACCUMULATOR_SIZE),
+		mainDataChannel:               make(chan *evData.EventData),
+		containersData: containersAccumulator{
+			accumulatorDataPerContainer: make(map[string]chan evData.EventData),
+		},
+	}
+
+	return accumulatorSingelton
+}
+
+func GetAccumulator() *Accumulator {
+	if accumulatorSingelton == nil {
```

Comment:
```suggestion
	if accumulatorSingelton != nil {
	    return accumulatorSingelton
	}
```
[Required]: This is the correct way

Reply:
this is how implementing a singleton, please see this link:
https://refactoring.guru/design-patterns/singleton/go/example